test/lib/y2firewall/clients/auto_test.rb

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env rspec
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2017 SUSE LLC
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE.
+#
+# To contact SUSE about this file by physical or electronic mail, you may find
+# current contact information at www.suse.com.
+# ------------------------------------------------------------------------------
+
+require_relative "../../../test_helper"
+require "y2firewall/clients/auto"
+
+describe Y2Firewall::Clients::Auto do
+  let(:firewalld) { Y2Firewall::Firewalld.instance }
+  let(:importer) { double("Y2Firewall::Importer", import: true) }
+
+  before do
+    allow(firewalld).to receive(:read)
+    allow(subject).to receive(:importer).and_return(importer)
+  end
+
+  describe "#summary" do
+    it "returns the summary of all the configured zones" do
+      expect(firewalld.api).to receive(:list_all_zones).and_return(["zone1", "zone2"])
```

Comment:
I would say that it is a "precondition", but not an expectation at all. I would switch it to `allow` (actually I would move it to a before block, but maybe that's too much).